src/lib/import-export/export/exporters/default-exporter.ts

```diff
@@ -223,10 +223,17 @@ export class DefaultExporter extends EventEmitter implements Exporter {
 		return directoryContents.reduce< string[] >( ( files: string[], directoryContent ) => {
 			const filePath = path.join( directoryContent.path, directoryContent.name );
 			const relativePath = path.relative( this.options.site.path, filePath );
+
+			// Check for exact path exclusions
 			const isExcluded = this.pathsToExclude.some( ( pathToExclude ) =>
 				relativePath.startsWith( path.normalize( pathToExclude ) )
 			);
-			if ( isExcluded ) {
+
+			// Check for node_modules and .git directories anywhere
+			const isNodeModulesDirectory = relativePath.includes( 'node_modules' );
+			const isGitDirectory = relativePath.includes( '.git' );
```

Comment:
I think it should be safe to exclude all those paths that include `node_modules` or `.git` in them, but happy to make it more strict if you think it's necessary. 🙂